community/primitive-collections/src/main/java/org/neo4j/concurrent/RecentK.java

```diff
@@ -30,116 +33,101 @@
  */
 public class RecentK<Type> implements Iterable<Type>
 {
+    /** Cut a number in half. */
+    public static final LongUnaryOperator COUNTER_HALFLIFE = new LongUnaryOperator()
```

Comment:
Maybe we can inline this constant? I believe it would be more readable and won't cause any performance hit